src/polysolve/LinearSolverSympiler.cpp

```diff
@@ -0,0 +1,82 @@
+#ifdef POLYSOLVE_WITH_SYMPILER
+
+#include <polysolve/LinearSolverSympiler.hpp>
+#include <Eigen/Sparse>
+#include <Eigen/Core>
+
+namespace polysolve
+{
+    void LinearSolverSympiler::setSolverMode(int _solver_mode)
+    {
+        solver_mode = _solver_mode;
+    }
+
+    void LinearSolverSympiler::setParameters(const json &params)
+    {
+        if (params.contains("Sympiler"))
+        {
+            if (params["Sympiler"].contains("solver_mode"))
+            {
+                setSolverMode(params["Sympiler"]["mtype"].get<int>());
+            }
+        }
+    }
+
+    void LinearSolverSympiler::getInfo(json &params) const
+    {
+        if (factorize_status == 1)
+        {
+            params["factorize_info"] = "Success";
+        }
+        else
+        {
+            params["factorize_info"] = "Failure";
+        }
+    }
+
+    void LinearSolverSympiler::analyzePattern(const StiffnessMatrix &A, const int precond_num)
+    {
+        StiffnessMatrix A_copy = StiffnessMatrix(A);
+
+        A_csc_->nzmax = A_copy.nonZeros();
+        A_csc_->ncol = A_csc_->nrow = A_copy.rows();
+        A_csc_->p = A_copy.outerIndexPtr();
+        A_csc_->i = A_copy.innerIndexPtr();
+
+        A_csc_->x = A_copy.valuePtr();
+        A_csc_->stype = -1;
+        A_csc_->packed = 1;
+        A_csc_->nz = nullptr;
+
+        solver_ = new sym_lib::parsy::SolverSettings(A_csc_);
+        solver_->symbolic_analysis();
+    }
+
+    // Factorize system matrix
+    void LinearSolverSympiler::factorize(const StiffnessMatrix &A)
```

Comment:
There is significant code duplication between `analyzePattern` and `factorize`. i.e. the initialization of `A_csc_` object. I think it can be factored out.

Reply:
Factored out the duplicated code into separate function